rest/views.py

```diff
@@ -819,7 +838,17 @@ def make_facet_counts(self, queryset):
             queryset = queryset.facet(vocabulary.id)
 
         facet_counts = queryset.facet_counts()
-        ret = get_vocabularies(facet_counts)
+        vocabs = get_vocabularies(facet_counts)
+
+        # return dictionary
+        ret_dict = {}
+
+        # process vocabularies
+        for key, values in vocabs.items():
+            missing_count = queryset.filter(
```

Comment:
I did not measure the difference in performance, but right now we don't have another choice, so unless the website becomes unusable, I guess this is the way to go.